tests/test_dojo.py

```diff
@@ -0,0 +1,264 @@
+            [{'testoffice': ['DELE ALI']}, {'testlivingspace': ['DELE ALI']}], result)
+
+    def test_tabular_output_on_print_allocations(self):
+        """Tests that tabular data is output on print_allocations"""
+
+        #Create StringIO object and redirect output
+        self.dojo.add_person("Dele", "Ali", "Fellow", "Y")
+        program_captured_output = io.StringIO()
+        sys.stdout = program_captured_output
+        self.dojo.print_allocations("", "Y")
+        sys.stdout = sys.__stdout__
+        table = PrettyTable(['Name', 'Type', 'Office', 'Living Space'])
+        table.add_row(["Dele Ali", "fellow", "testoffice", "testlivingspace"])
+
+        captured_output = io.StringIO()
+        sys.stdout = captured_output
+        print(colorful.blue("List showing people with space and their respective rooms"))
+        print(colorful.blue(table))
+        sys.stdout = sys.__stdout__
+        print(program_captured_output.getvalue().strip())
+        print(captured_output.getvalue())
+        self.assertTrue(
+            captured_output.getvalue().strip() in program_captured_output.getvalue().strip())
+
+    def test_tabular_output_on_print_unallocated(self):
+        """Tests that tabular data is output on test_tabular_output_on_print_unallocated"""
+
+        dojo = Dojo()
+        dojo.add_person("Kylian", "Mbappe", "Fellow", "Y")
+        dojo.add_person("Gianluggi", "Buffon", "Fellow", "N")
+        dojo.create_room("office", "red")
+        dojo.add_person("Timoue", "Bakayoko", "Fellow", "Y")
+        program_captured_output = io.StringIO()
+        sys.stdout = program_captured_output
+        dojo.print_unallocated()
+        sys.stdout = sys.__stdout__
+        table = PrettyTable(['Name', 'Person id', 'Missing'])
+        table.add_row(["Kylian Mbappe", "1", "Office and Living Space"])
+        table.add_row(["Gianluggi Buffon", "2", "Office"])
+        table.add_row(["Timoue Bakayoko", "3", "Living Space"])
+
+        captured_output = io.StringIO()
+        sys.stdout = captured_output
+        print(colorful.blue("Table showing people along with missing rooms"))
+        print(colorful.blue(table))
+        sys.stdout = sys.__stdout__
+        print(program_captured_output.getvalue().strip())
+        print(captured_output.getvalue())
+        self.assertTrue(
+            captured_output.getvalue().strip() in program_captured_output.getvalue().strip())
+
+    def test_reallocate_if_person_had_no_office(self):
+        """Tests reallocate if person had no office"""
+
+        dojo = Dojo()
+        dojo.add_person("John", "Ashaba", "Staff", "Y")
+        dojo.create_room("office", "orange")
+        dojo.reallocate_person(1, "orange")
+        target_room = dojo.find_room("orange")
+        person = dojo.find_person(1)
+        self.assertIn(person, target_room.residents)
+
+    def test_reallocate_if_person_had_no_living_space(self):
+        """Tests reallocate if person had no living space"""
+
+        dojo = Dojo()
```

Comment:
You probably need not to duplicate this instance and make use of what you already have in the setup.